gateway/gateway.go

```diff
@@ -146,6 +148,13 @@ func (g *gateway) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 		g.started = false
 	}
 
+	// capture C log output
+	loggingStarted, ok := LoggingRoutineStarted[g.Name().Name]
+	if !ok || !loggingStarted {
+		g.workers = utils.NewBackgroundStoppableWorkers(g.captureCOutputToLogs)
+		LoggingRoutineStarted[g.Name().Name] = true
+	}
+
```

Comment:
I can't add two resources with the same name from the resource graph, but I can with the exported constructor function call, can we ensure that the Logger map is overwritten in the correct way, and also make this logic into a struct with functions that are testable, and write tests for them to ensure that we append the logger map in the correct way.